webkitgtk.yml

```diff
@@ -0,0 +1,181 @@
+        sha256: 398fe7039ce35db80fe7da8d116035924f2c02ea4a4aa9f4903df6699287599c
+        x-checker-data:
+          type: json
+          url: https://api.github.com/repos/AOMediaCodec/libavif/releases/latest
+          tag-query: .tag_name
+          timestamp-query: .published_at
+          version-query: $tag | sub("^[vV]"; "")
+          url-query: '"https://github.com/AOMediaCodec/libavif/archive/\($tag)/libavif-\($version).tar.gz"'
+    cleanup:
+      - '*'
+  - name: libjxl
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      - -DBUILD_TESTING=OFF
+      - -DJPEGXL_ENABLE_DEVTOOLS=OFF
+      - -DJPEGXL_ENABLE_TOOLS=OFF
+      - -DJPEGXL_ENABLE_DOXYGEN=OFF
+      - -DJPEGXL_ENABLE_MANPAGES=OFF
+      - -DJPEGXL_ENABLE_BENCHMARK=OFF
+      - -DJPEGXL_ENABLE_EXAMPLES=OFF
+      - -DJPEGXL_ENABLE_JNI=OFF
+      - -DJPEGXL_ENABLE_SJPEG=OFF
+      - -DJPEGXL_ENABLE_OPENEXR=OFF
+      - -DJPEGXL_ENABLE_SKCMS=OFF
+    sources:
+      - type: archive
+        url: https://github.com/libjxl/libjxl/archive/v0.8.2/libjxl-0.8.2.tar.gz
+        sha256: c70916fb3ed43784eb840f82f05d390053a558e2da106e40863919238fa7b420
+        x-checker-data:
+          type: json
+          url: https://api.github.com/repos/libjxl/libjxl/releases/latest
+          tag-query: .tag_name
+          timestamp-query: .published_at
+          version-query: $tag | sub("^[vV]"; "")
+          url-query: '"https://github.com/libjxl/libjxl/archive/\($tag)/libjxl-\($version).tar.gz"'
+    cleanup:
+      - '*'
+  - name: libwpe
+    buildsystem: meson
+    builddir: true
+    config-opts:
+      - --buildtype=plain
+    sources:
+      - type: archive
+        url: https://github.com/WebPlatformForEmbedded/libwpe/releases/download/1.14.1/libwpe-1.14.1.tar.xz
+        sha256: b1d0cdcf0f8dbb494e65b0f7913e357106da9a0d57f4fbb7b9d1238a6dbe9ade
+        x-checker-data:
+          type: json
+          url: https://api.github.com/repos/WebPlatformForEmbedded/libwpe/releases/latest
+          tag-query: .tag_name
+          timestamp-query: .published_at
+          version-query: $tag
+          url-query: '"https://github.com/WebPlatformForEmbedded/libwpe/releases/download/\($tag)/libwpe-\($version).tar.xz"'
+  - name: wpebackend-fdo
+    buildsystem: meson
+    builddir: true
+    config-opts:
+      - --buildtype=plain
+    sources:
+      - type: archive
+        url: https://github.com/Igalia/WPEBackend-fdo/releases/download/1.14.2/wpebackend-fdo-1.14.2.tar.xz
+        sha256: 93c9766ae9864eeaeaee2b0a74f22cbca08df42c1a1bdb55b086f2528e380d38
+        x-checker-data:
+          type: json
+          url: https://api.github.com/repos/Igalia/WPEBackend-fdo/releases/latest
+          tag-query: .tag_name
+          timestamp-query: .published_at
+          version-query: $tag
+          url-query: '"https://github.com/Igalia/WPEBackend-fdo/releases/download/\($tag)/wpebackend-fdo-\($version).tar.xz"'
```

Comment:
Also all of these are in the `org.gnome.Platform` runtime so you could switch to that to avoid maintaining these.

Though FDO is a bit smaller in general so up to you.

Reply:
I won't deny that there's a philosophical angle in sticking with FDO. Geany is a GTK3 app, and makes a point of avoiding most GNOME-specific technologies in order to provide an equal experience on all desktops. I feel that it's most appropriate for desktop-agnostic apps to target the FDO runtime, but much of that is admittedly philosophical, ideological, etc.

There are tangible benefits, too; FDO is not only smaller, but also releases once a year, as opposed to the twice a year release schedule of GNOME. That also means as it currently is, if I get bus factored, it wouldn't be EOL until 25.08 releases.

All that said, I won't rule out switching to the GNOME runtime if it turns out that these components have a lot of churn and cause flathubbot to make a bunch of noise all the time.